src/torchmetrics/audio/srmr.py

```diff
@@ -0,0 +1,162 @@
+
+    - ``preds`` (:class:`~torch.Tensor`): float tensor with shape ``(...,time)``
+
+    As output of `forward` and `compute` the metric returns the following output
+
+    - ``srmr`` (:class:`~torch.Tensor`): float scaler tensor
+
+    .. note:: using this metrics requires you to have ``gammatone`` and ``torchaudio`` installed.
+        Either install as ``pip install torchmetrics[audio]`` or ``pip install torchaudio``
+        and ``pip install git+https://github.com/detly/gammatone``.
+
+    Args:
+        preds: shape ``(..., time)``
+        fs: the sampling rate
+        n_cochlear_filters: Number of filters in the acoustic filterbank
+        low_freq: determines the frequency cutoff for the corresponding gammatone filterbank.
+        min_cf: Center frequency in Hz of the first modulation filter.
+        max_cf: Center frequency in Hz of the last modulation filter. If None is given,
+            then 30 Hz will be used for `norm==False`, otherwise 128 Hz will be used.
+        norm: Use modulation spectrum energy normalization
+        fast: Use the faster version based on the gammatonegram.
+            Note: this argument is inherited from `SRMRpy`_. As the translated code is based to pytorch,
+            setting `fast=True` may slow down the speed for calculating this metric on GPU.
+
+    Raises:
+        ModuleNotFoundError:
+            If ``gammatone`` or ``torchaudio`` package is not installed
+
+    Example:
+        >>> import torch
+        >>> from torchmetrics.audio import SpeechReverberationModulationEnergyRatio
+        >>> g = torch.manual_seed(1)
+        >>> preds = torch.randn(8000)
+        >>> srmr = SpeechReverberationModulationEnergyRatio(8000)
+        >>> srmr(preds)
+        tensor(0.3354)
+    """
+
+    msum: Tensor
+    total: Tensor
+    full_state_update: bool = False
+    is_differentiable: bool = True
+    higher_is_better: bool = True
+    plot_lower_bound: Optional[float] = None
+    plot_upper_bound: Optional[float] = None
+
+    def __init__(
+        self,
+        fs: int,
+        n_cochlear_filters: int = 23,
+        low_freq: float = 125,
+        min_cf: float = 4,
+        max_cf: Optional[float] = 128,
+        norm: bool = False,
+        fast: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        if not _TORCHAUDIO_AVAILABEL or not _GAMMATONE_AVAILABEL:
+            raise ModuleNotFoundError(
+                "speech_reverberation_modulation_energy_ratio requires you to have `gammatone` and"
+                " `torchaudio` installed. Either install as ``pip install torchmetrics[audio]`` or "
+                "``pip install torchaudio`` and ``pip install git+https://github.com/detly/gammatone``"
+            )
+        self.fs = fs
+        self.n_cochlear_filters = n_cochlear_filters
+        self.low_freq = low_freq
+        self.min_cf = min_cf
+        self.max_cf = max_cf
+        self.norm = norm
+        self.fast = fast
```

Comment:
please also add input validation for the rest :)

Reply:
added